src/button-group/icon-button-item.tsx

```diff
@@ -0,0 +1,94 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useRef, useState, forwardRef } from 'react';
+import { ButtonGroupProps } from './interfaces.js';
+import { InternalButton } from '../button/internal.js';
+import { ClickDetail, fireCancelableEvent } from '../internal/events/index.js';
+import { ButtonProps } from '../button/interfaces.js';
+import Tooltip from './tooltip/index.js';
+import StatusIndicator from '../status-indicator/internal.js';
+import styles from './styles.css.js';
+
+const IconButtonItem = forwardRef(
+  (
+    {
+      item,
+      onItemClick,
+    }: {
+      item: ButtonGroupProps.IconButton;
+      onItemClick?: (event: CustomEvent) => void;
+    },
+    ref: React.Ref<ButtonProps.Ref>
+  ) => {
+    const buttonRef = useRef<HTMLDivElement>(null);
+    const [popoverOpen, setPopoverOpen] = useState(false);
+    const [isActionPopover, setIsActionPopover] = useState(false);
+    const hasIcon = item.iconName || item.iconUrl || item.iconSvg;
+
+    const onClickHandler = (event: CustomEvent<ClickDetail>) => {
+      if (item.actionPopoverText) {
+        setIsActionPopover(true);
+        setPopoverOpen(true);
+      } else {
+        setPopoverOpen(false);
+      }
+
+      if (onItemClick) {
```

Comment:
This IF condition is redundant - the fireCancellableEvent accepts nullable handler.